db/migrate/20231210061330_add_created_at_to_fts_targets_with_index.rb

```diff
@@ -0,0 +1,109 @@
+# For auto load
+FullTextSearch::Migration
+
+class AddCreatedAtToFtsTargetsWithIndex < ActiveRecord::Migration[5.2]
+  def up
+    return if !table_exists?(:fts_targets)
+
+    add_column :fts_targets, :created_at, :timestamp
+
+    ActiveRecord::Base.transaction do
+      update_created_at_using_last_modified_at
+      update_created_at_using_created_on
+    end
+
+    if Redmine::Database.mysql?
+      add_index :fts_targets, :created_at
+    else
+      remove_index :fts_targets, name: "fts_targets_index_pgroonga"
+      add_index :fts_targets,
+                [:id,
+                 :source_id,
+                 :source_type_id,
+                 :project_id,
+                 :container_id,
+                 :container_type_id,
+                 :custom_field_id,
+                 :is_private,
+                 :last_modified_at,
+                 :created_at,
+                 :title,
+                 :content,
+                 :tag_ids],
+                using: "PGroonga",
+                with: "normalizer = 'NormalizerNFKC121'",
+                name: "fts_targets_index_pgroonga"
+    end
+  end
+
+  def down
+    return if !table_exists?(:fts_targets)
+
+    if Redmine::Database.mysql?
+      remove_index :fts_targets, :created_at
+    else
+      remove_index :fts_targets, name: "fts_targets_index_pgroonga"
+      add_index :fts_targets,
+                [:id,
+                 :source_id,
+                 :source_type_id,
+                 :project_id,
+                 :container_id,
+                 :container_type_id,
+                 :custom_field_id,
+                 :is_private,
+                 :last_modified_at,
+                 :title,
+                 :content,
+                 :tag_ids],
+                using: "PGroonga",
+                with: "normalizer = 'NormalizerNFKC121'",
+                name: "fts_targets_index_pgroonga"
+    end
+
+    remove_column :fts_targets, :created_at
+  end
+
+  private
+
+  class FtsTarget < ActiveRecord::Base
+  end
+
+  class FtsType < ActiveRecord::Base
+  end
+
+  class Issue < ActiveRecord::Base
+  end
+
+  class Message < ActiveRecord::Base
+  end
+
+  class Project < ActiveRecord::Base
+  end
+
+  class WikiPage < ActiveRecord::Base
+  end
+
+  def update_created_at_using_last_modified_at
+    types = ['Attachment', 'Change', 'Changeset', 'CustomValue', 'Document', 'Journal', 'News']
+
+    FtsTarget.where(source_type_id: FtsType.where(name: types).select(:id))
+             .update_all(created_at: FtsTarget.arel_table[:last_modified_at])
+  end
+
+  def update_created_at_using_created_on
+    types = ['Issue', 'Message', 'Project', 'WikiPage']
```

Comment:
Can we use class instead of class name?

```suggestion
    types = [Issue, Message, Project, WikiPage]
```

Reply:
fix: [285a2bf](https://github.com/clear-code/redmine_full_text_search/pull/118/commits/285a2bf64d72b35621bf7598cc279b2034af16a7), [4e577ac](https://github.com/clear-code/redmine_full_text_search/pull/118/commits/4e577acb6ff717c3b677c175b956781549bef5ab)